### PR TITLE
aarch64: CPU branding info and misc improvements

### PR DESCRIPTION
**Pull request by kd-11**
- Add some basic CPU brand name detection. I do not have a complete list, but we have enough to get started.
- Fix macOS arm64 compilation.

Closes https://github.com/RPCS3/rpcs3/issues/15980
Closes https://github.com/RPCS3/rpcs3/issues/15972

### CODE REVIEW

File: rpcs3/Emu/CPU/Backends/AArch64/AArch64Signal.cpp [hunk 1]
```diff
@@ -47,16 +48,29 @@ namespace aarch64
...
 
-    fault_reason decode_fault_reason(const ucontext_t* uctx)
+    u64 _read_ESR_EL1(const ucontext_t* uctx)
     {
         auto esr_ctx = find_EL1_esr_context(uctx);
-        if (!esr_ctx)
+        return esr_ctx ? esr_ctx->esr : 0;
+    }
+#else
+    u64 _read_ESR_EL1(const ucontext_t* uctx)
```
Comment (Megamouse) [inline rpcs3/Emu/CPU/Backends/AArch64/AArch64Signal.cpp:57]: Unused param

File: rpcs3/Emu/CPU/Backends/AArch64/AArch64Common.cpp [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        { 0x41, 0xd49, "armv8.5-a+fp16+bf16+i8mm", "", "Neoverse-N2" },
+        { 0x41, 0xd23, "armv8.1-m.main+pacbti+mve.fp+fp.dp", "", "Cortex-M85" },
+        { 0x41, 0xd13, "armv8-r+crc+simd", "", "Cortex-R52" },
+        { 0x41, 0xd16, "armv8-r+crc+simd", "", "Cortex-R52+" },
+
+        // APPLE
+        { 0x61, 0x22, "armv8.5-a", "M1", "Firestorm" },
+        { 0x61, 0x23, "armv8.5-a", "M1", "IceStorm" },
+        { 0x61, 0x28, "armv8.5-a", "M1 Max", "Firestorm" },
+        { 0x61, 0x28, "armv8.5-a", "M1 Max", "Icestorm" },
```
Comment (Megamouse) [inline rpcs3/Emu/CPU/Backends/AArch64/AArch64Common.cpp:76]: How does this differ from the one above?

File: rpcs3/Emu/CPU/Backends/AArch64/AArch64Common.cpp [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        { 0x61, 0x24, "armv8.5-a", "M1 Pro", "Firestorm" },
+        { 0x61, 0x25, "armv8.5-a", "M1 Pro", "Icestorm" },
+        { 0x61, 0x32, "armv8.5-a", "M2", "Avalanche" },
+        { 0x61, 0x32, "armv8.5-a", "M2", "Blizzard" },
```
Comment (Megamouse) [inline rpcs3/Emu/CPU/Backends/AArch64/AArch64Common.cpp:80]: Same here

File: rpcs3/Emu/CPU/Backends/AArch64/AArch64Common.cpp [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+            {
+                return &cpu;
+            }
+        }
+
+        return nullptr;
+    }
+
+    // Read main ID register
+    static u64 read_MIDR_EL1(u32 cpu_id)
```
Comment (Megamouse) [inline rpcs3/Emu/CPU/Backends/AArch64/AArch64Common.cpp:113]: Unused param

File: rpcs3/Emu/CPU/Backends/AArch64/AArch64Common.cpp [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    {
+#if defined(__linux__)
+        std::string path = fmt::format("/sys/devices/system/cpu/cpu%u/regs/identification/midr_el1", cpu_id);
```
Comment (Megamouse) [inline rpcs3/Emu/CPU/Backends/AArch64/AArch64Common.cpp:116]: const

**Review by hilop27 — APPROVED** [review]

**Pull request merged**